rust/osm2lanes/src/lib.rs

```diff
@@ -174,6 +308,46 @@ mod tests {
         rust: Option<RustTesting>,
     }
 
+    impl Road {
+        fn approx_eq(&self, other: &Self) -> bool {
```

Comment:
How about a docstring: `/// Marking color only counts for equality when defined on both objects being compared`